test/check-machines-settings

```diff
@@ -384,6 +384,8 @@ class TestMachinesSettings(VirtualMachinesCase):
         b.wait_in_text("#vm-subVmTest1-state", "Running")
         self.goToVmPage("subVmTest1")
 
+        wait(lambda: "login as 'cirros' user" in m.execute("cat {0}".format(args["logfile"])), delay=3)
```

Comment:
Oh -- changing CPUs while the kernel is still booting might screw up too much in the guest? (although I didn't expect that we actually care what's running inside). But if it just slows down the test enough, that's fine :grin:

Reply:
this is actually the commit that stabilized the test. the others are just test improvents and cleanups on the code. The 'why' part I can't answer :/

Reply:
Sure, let's leave some mysteries :grin: I'm fine with it, I was just thinking aloud what could happen here. But I suppose it's just a glorified sleep.

Reply:
What was happening in the test, is that the API call which was changing the CPU settings was just not taking effect - or IOW leaving the previous CPU settings values, and that without some error message.

So indeed there was some race when changing the XML of the just created domain. 

I think this scenario for the users to hit is impossible so I am also ok with this change.